### DISCUSSION TITLE
Fallback geolocation method(s)

### DISCUSSION THREAD

**Discussion by nitefood** [GyulyVGC/sniffnet#210] [Ideas]
I reckon the speedup improvement that comes from using local MMDB-based geolocation is priceless, but the trade-off is that its coverage will obviously never be 100%. For that reason, I would suggest using a fallback API-based geolocation lookup (or two) that could fill in some of the holes left.
Of course that introduces some latency and may be require some sort of background processing in order not to interefere with realtime monitoring responsiveness, but I guess users may be informed of this and be able to choose to explicitly enable this feature in the settings pane.

Some geolocation providers that do not require an API key (which I guess is a unnecessary burden for most end users) while at the same time providing good precision are [RIPE IPmap](https://ipmap.ripe.net/) and [ip-api](https://ip-api.com/).

**Comment by GyulyVGC** [comment]
Thanks for the suggestion and resources, your advices are really precious.
I'm already performing rDNS lookups as a background activity (besides other), so I may consider using the same threads also for this job.

**Reply by nitefood** [reply]
You're welcome! Keep up the great work!

**Comment by DarkSky7** [comment]
I wish for the ability to select multiple interfaces (e.g. LAN & WiFi) and plot combined bandwidth, inbound(LAN + WiFi) and outbound (LAN + WiFi) because the interface over which information moves is irrelevant; the complete story is what's interesting.
